src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/dynamicProgramming/WrappingDPAlgorithm.java

```diff
@@ -0,0 +1,63 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.dynamicProgramming;
+
+import se.liu.ida.hefquin.engine.federation.FederationAccessException;
+import se.liu.ida.hefquin.engine.queryplan.LogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpMultiwayJoin;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperatorForLogicalOperator;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.QueryOptimizationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class WrappingDPAlgorithm {
+    protected final QueryOptimizationContext ctxt;
+
+    public WrappingDPAlgorithm( final QueryOptimizationContext ctxt ) {
+        assert ctxt != null;
+        this.ctxt = ctxt;
+    }
+
+    public PhysicalPlan optimize( final LogicalPlan initialPlan ) throws FederationAccessException, QueryOptimizationException {
+        final boolean keepMultiwayJoins = true;
+        final PhysicalPlan initialPhysicalPlan = ctxt.getLogicalToPhysicalPlanConverter().convert(initialPlan, keepMultiwayJoins);
+
+        return optimizePhysicalPlan(initialPhysicalPlan);
+    }
+
+    public PhysicalPlan optimizePhysicalPlan( final PhysicalPlan pp) throws FederationAccessException, QueryOptimizationException {
+        final PhysicalOperatorForLogicalOperator pop = (PhysicalOperatorForLogicalOperator) pp.getRootOperator();
+        final LogicalOperator lop = pop.getLogicalOperator();
+        if ( lop instanceof LogicalOpMultiwayJoin) {
+            final List<PhysicalPlan> children = findChildren(pp);
+            return rewrite( children );
+        }
+        else {
+            return pp;
+        }
+    }
+
+    public List<PhysicalPlan> findChildren( final PhysicalPlan pp ) throws FederationAccessException, QueryOptimizationException {
+        final List<PhysicalPlan> children = new ArrayList<PhysicalPlan>();
+        final int numChildren = pp.numberOfSubPlans();
+        if ( numChildren > 0 ) {
+            for ( int i = 0; i < numChildren; ++i ) {
+                children.add( optimizePhysicalPlan(pp.getSubPlan(i)) );
+            }
+        }
+        return children;
+    }
+
+    public PhysicalPlan rewrite( final List<PhysicalPlan> children ) throws QueryOptimizationException, FederationAccessException {
+        if ( children.size() < 1 )
+            throw new IllegalArgumentException( "unexpected number of sub-plans: " + children.size() );
+        else if (children.size() == 1 ){
```

Comment:
Can you please try to be consistent in how you are formatting your code. For instance, this else-if branch is wrapped in curly braces and appears on a new line, whereas the if branch directly above does not have curly braces ... but at least it also is on a new, which is not always the case in this and the other PR.  Another example: in this line here you have no space between the opening parenthesis and the if condition, but two lines before you do.